subsys/shell/backends/shell_telnet.c

```diff
@@ -197,22 +191,25 @@ static int telnet_send(void)
 		return 0;
 	}
 
-	if (sh_telnet->client_ctx == NULL) {
+	if (sh_telnet->fds[SOCK_ID_CLIENT].fd < 0) {
 		return -ENOTCONN;
 	}
 
 	while (len > 0) {
-		ret = net_context_send(sh_telnet->client_ctx, msg,
-				       len, telnet_sent_cb,
-				       K_FOREVER, NULL);
-
-		if (ret == -EAGAIN || ret == -ENOBUFS) {
-			k_sleep(K_MSEC(TELNET_RETRY_SEND_SLEEP_MS));
-			continue;
+		ret = zsock_send(sh_telnet->fds[SOCK_ID_CLIENT].fd, msg, len,
+				 block ? 0 : ZSOCK_MSG_DONTWAIT);
+		if (!block && (ret < 0) & (errno == EAGAIN)) {
```

Comment:
I think `&&` was intended?

```suggestion
		if (!block && (ret < 0) && (errno == EAGAIN)) {
```

Reply:
True, fixed